rust/yarp-sys/build.rs

```diff
@@ -0,0 +1,180 @@
+        .allowlist_type("yp_string_list_t") // TEST ✅
+        .allowlist_type("yp_token_type_t")
+        // Enums
+        .rustified_non_exhaustive_enum("yp_comment_type_t") // TEST ✅
+        .rustified_non_exhaustive_enum("yp_context_t")
+        .rustified_non_exhaustive_enum("yp_heredoc_indent_t")
+        .rustified_non_exhaustive_enum("yp_heredoc_quote_t")
+        .rustified_non_exhaustive_enum("yp_lex_mode_t")
+        .rustified_non_exhaustive_enum("yp_lex_state_t")
+        .rustified_non_exhaustive_enum("yp_node_type_t") // TEST ✅
+        .rustified_non_exhaustive_enum("yp_pack_encoding") // TEST ✅
+        .rustified_non_exhaustive_enum("yp_pack_endian") // TEST ✅
+        .rustified_non_exhaustive_enum("yp_pack_length_type") // TEST ✅
+        .rustified_non_exhaustive_enum("yp_pack_result") // TEST ✅
+        .rustified_non_exhaustive_enum("yp_pack_signed") // TEST ✅
+        .rustified_non_exhaustive_enum("yp_pack_size") // TEST ✅
+        .rustified_non_exhaustive_enum("yp_pack_type") // TEST ✅
+        .rustified_non_exhaustive_enum("yp_pack_variant") // TEST ✅
+        .rustified_non_exhaustive_enum("yp_token_type")
+        .rustified_non_exhaustive_enum("yp_unescape_type_t") // TEST ✅
+        // Functions
+        .allowlist_function("yp_buffer_init") // TEST ✅
+        .allowlist_function("yp_buffer_free") // TEST ✅
+        .allowlist_function("yp_node_destroy") // TEST ✅
+        .allowlist_function("yp_list_empty_p") // TEST ✅
+        .allowlist_function("yp_list_free") // TEST ✅
+        .allowlist_function("yp_list_init") // TEST ✅
+        .allowlist_function("yp_node_memsize") // TEST ✅
+        .allowlist_function("yp_pack_parse") // TEST ✅
+        .allowlist_function("yp_parse") // TEST ✅
+        .allowlist_function("yp_parse_serialize") // TEST ✅
+        .allowlist_function("yp_parser_free") // TEST ✅
+        .allowlist_function("yp_parser_init") // TEST ✅
+        .allowlist_function("yp_parser_register_encoding_changed_callback") // TEST ✅
+        .allowlist_function("yp_parser_register_encoding_decode_callback") // TEST ✅
+        .allowlist_function("yp_prettyprint") // TEST ✅
+        .allowlist_function("yp_regexp_named_capture_group_names")
+        .allowlist_function("yp_serialize") // TEST ✅
+        .allowlist_function("yp_size_to_native") // TEST ✅
+        .allowlist_function("yp_string_free") // TEST ✅
+        .allowlist_function("yp_string_length") // TEST ✅
+        .allowlist_function("yp_string_source") // TEST ✅
+        .allowlist_function("yp_string_list_init") // TEST ✅
+        .allowlist_function("yp_string_list_free") // TEST ✅
+        .allowlist_function("yp_token_type_to_str")
+        .allowlist_function("yp_unescape_calculate_difference") // TEST ✅
+        .allowlist_function("yp_unescape_manipulate_string") // TEST ✅
+        .allowlist_function("yp_version") // TEST ✅
+        // Vars
+        .allowlist_var("yp_encoding_ascii")
```

Comment:
These vars get turned into `static` vars, which seemed nice to have if you're doing anything with those encoding callbacks, but again I'm not sure if they should be in or not.
<img width="253" alt="image" src="https://github.com/ruby/yarp/assets/142010/b1639704-7bfe-49eb-a283-526188965d27">

Reply:
[ ] - Look into using a regex for these (to capture newly added encodings in the future